frame/nfts/src/features/lock.rs

```diff
@@ -31,15 +31,18 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		CollectionConfigOf::<T, I>::try_mutate(collection, |maybe_config| {
 			let config = maybe_config.as_mut().ok_or(Error::<T, I>::NoConfig)?;
 
-			if lock_config.has_disabled_setting(CollectionSetting::TransferableItems) {
+			if lock_settings.is_disabled(CollectionSetting::TransferableItems) {
```

Comment:
I still dont understand the code here. Why are there so many `ifs`?

Reply:
Refactored (twice :D)